prediction_market_agent/tools/mech/utils.py

```diff
@@ -0,0 +1,92 @@
+    prediction_with_research_report,
+)
+from prediction_market_agent.tools.mech.mech.packages.valory.customs.prediction_request import (
+    prediction_request,
+)
+from prediction_market_agent.utils import completion_str_to_json
+
+
+class MechResult(BaseModel):
+    p_yes: float
+    p_no: float
+    confidence: float
+    info_utility: float
+
+
+@contextmanager
+def saved_str_to_tmpfile(s: str) -> t.Iterator[str]:
+    # Write the string to the temporary file
+    with tempfile.NamedTemporaryFile(delete=False) as tmp:
+        tmp.write(s.encode())
+
+    yield tmp.name
+
+    # Finally remove the temporary file
+    os.remove(tmp.name)
+
+
+class MechTool(str, Enum):
+    PREDICTION_WITH_RESEARCH_REPORT = "prediction-with-research-conservative"
+    PREDICTION_ONLINE = "prediction-online"
+
+
+def mech_request(question: str, mech_tool: MechTool) -> MechResult:
+    private_key = MechAPIKeys().bet_from_private_key.get_secret_value()
+    with saved_str_to_tmpfile(private_key) as tmpfile_path:
+        # Increase gas price to reduce chance of 'out of gas' transaction failures
+        mech_strategy_env_var = "MECHX_LEDGER_DEFAULT_GAS_PRICE_STRATEGY"
+        if os.getenv(mech_strategy_env_var):
+            raise ValueError(f"{mech_strategy_env_var} already set in the environment.")
+        os.environ[mech_strategy_env_var] = "gas_station"
+
+        response = interact(
+            prompt=question,
+            # Taken from https://github.com/valory-xyz/mech?tab=readme-ov-file#examples-of-deployed-mechs
+            agent_id=6,
+            private_key_path=tmpfile_path,
+            # To see a list of available tools, comment out the tool parameter
+            # and run the function. You will be prompted to select a tool.
+            tool=mech_tool.value,
+            confirmation_type=ConfirmationType.WAIT_FOR_BOTH,
+        )
+        del os.environ[mech_strategy_env_var]
+        return MechResult.model_validate_json(response["result"])
+
+
+def mech_request_local(question: str, mech_tool: MechTool) -> MechResult:
+    keys = MechAPIKeys()
+    if mech_tool == MechTool.PREDICTION_WITH_RESEARCH_REPORT:
+        response = prediction_with_research_report.run(
+            tool=mech_tool.value,
+            prompt=question,
+            api_keys={
+                "openai": keys.openai_api_key.get_secret_value(),
+                "tavily": keys.tavily_api_key.get_secret_value(),
+            },
+        )
+    elif mech_tool == MechTool.PREDICTION_ONLINE:
+        response = prediction_request.run(
+            tool=mech_tool.value,
+            prompt=question,
+            api_keys={
+                "openai": keys.openai_api_key.get_secret_value(),
+                "google_api_key": keys.google_search_api_key.get_secret_value(),
+                "google_engine_id": keys.google_search_engine_id.get_secret_value(),
+            },
+        )
```

Comment:
My understanding was that, if the method is called `run_..._local`, that a local resource would be used and no costs would rise from it.
When you call `prediction_request` (from the mech package), what happens?

Reply:
> no costs would rise from it.

Yes, no costs incurred via crypto wallet payments. But these functions generally have search and LLM api calls, so those costs are incurred.

If you call mechs via the client, you pay 0.01 in xdai and don't need openai/search api keys. If you run locally, you need to these api keys, but no crypto wallet.